sw/host/cryptotest/cryptotest_parser.rs

```diff
@@ -0,0 +1,904 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Parser for cryptolib test fixtures.
+//!
+//! This file consumes `.c` files and performs a very basic parse on them
+//! to extract the test vector type and any annotations on it.
+//!
+//! This library expects the `.c` file to contain a struct definition
+//! conforming to the following reduced grammar, modulo whitespace.
+//!
+//! ```text
+//! // cryptotest:struct
+//! // Comments...
+//! typedef struct $name? {
+//!   // Comments...
+//!   $type $field;
+//!   // More fields...
+//! } $name;
+//! ```
+//!
+//! Here, `// Comments...` is any number of single-line comments, `$field` is any
+//! valid C identifier, and `$type` is one of the following:
+//! - `bool`
+//! - `uint8_t`, `uint16_t`, `uint32_t`, `uint64_t`
+//! - `int8_t`, `int16_t`, `int32_t`, `int64_t`
+//! - `size_t`
+//! - `const char *` (always a NUL-terminated string).
+//!
+//! Additionally, the following compound types are recognized:
+//! - `Int field[N];`, where `Int` is an integer type above and `N` is a literal.
```

Comment:
Just checking my understanding here -- this means `Int` is one of `int8_t`, `int16_t`, `int32_t`, `int64_t`, right? As a nitpick, I might put a `$` or something in front of `Int` because originally I parsed it as part of the syntax of the compound type.